libs/ml/tests/unit/ops/activations/logsoftmax.cpp

```diff
@@ -79,7 +79,9 @@ TYPED_TEST(LogSoftmaxTest, forward_3d_tensor_axis_0_test)
   op.Forward({std::make_shared<const TensorType>(data)}, prediction);
 
   // test correct values
-  ASSERT_TRUE(prediction.AllClose(gt, DataType{1e-3f}, DataType{1e-3f}));
+  ASSERT_TRUE(
+      prediction.AllClose(gt, static_cast<DataType>(10000) * math::function_tolerance<DataType>(),
+                          static_cast<DataType>(10000) * math::function_tolerance<DataType>()));
```

Comment:
Isn't 10000 a bit high? It seems to be a factor of 10 greater than the other one in this file that's been converted from an absolute tolerance to function_tolerance